### PR TITLE
bootloader: zero-terminate readlink result, fixes #1635

### LINKED ISSUE #1635: bootloader & freebsd: exe location extraction from /proc broken
For binaries made with single-file mode and if /proc is mounted on FreeBSD (seems not all users have it mounted), it tries to get the full path from there, but there are some funny characters at the end:
```
root@muenchen:/ # borg
Cannot open self /usr/local/bin/borg▒▒ or archive /usr/local/bin/borg▒▒.pkg
```
Maybe some \0 missing?

See there, line 253:
https://github.com/pyinstaller/pyinstaller/commit/d2e749f0d1fede80828c29837940a9269f596ebb

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

File: bootloader/src/pyi_path.c [hunk 1]
```diff
@@ -254,6 +254,11 @@ int pyi_path_executable(char *execfile, const char *appname)
 #elif defined(__sun)
     result = readlink("/proc/self/path/a.out", execfile, PATH_MAX);  // Solaris
 #endif
+    if(result >= 0) {
+        /* execfile is not yet zero-terminated. result is the byte count. */
+        *(execfile + result) = '\0';
+    }
+    else
     if(-1 == result) {
```
Comment (htgoebel) [inline bootloader/src/pyi_path.c]: This looks weird. Doesn't C have a `elif`, too?

Comment (ThomasWaldmann) [inline bootloader/src/pyi_path.c]: Nope.

**Comment by ThomasWaldmann** [comment]
BTW, the landscape-bot results seem a bit strange. The stuff it complains about is unrelated to my PR and are not "new problems".

**Comment by htgoebel** [comment]
> BTW, the landscape-bot results seem a bit strange

Yes it is. Looks like it's not yet configured. it even complains about "illegal constant names".

**Comment by ThomasWaldmann** [comment]
ok, I addressed the issues that were found, let's see if it works now.

**Comment by ThomasWaldmann** [comment]
Note: the appveyor fails look like a test platform issue, not an issue within my changes.

**Comment by htgoebel** [comment]
@codewarrior0 Can you pleas take care of this - it's a simple one.

**Comment by codewarrior0** [comment]
LGTM!

**Pull request merged**